pypika_tortoise/terms.py

```diff
@@ -1768,6 +1768,11 @@ class Pow(Function):
     def __init__(self, term: Term, exponent: float, alias: str | None = None) -> None:
         super().__init__("POW", term, exponent, alias=alias)
 
+    def get_function_sql(self, ctx: SqlContext) -> str:
+        if ctx.dialect == Dialects.MSSQL:
+            self.name = "POWER"
```

Comment:
It's a hell of a side effect when `get_function_sql` changes the object state. You don't expect a getter to change the state.

I think we can make `name` a function accepting `SqlContext` in `Function`, and then override it in `Pow`.